com.unity.cinemachine/Editor/Upgrader/CinemachineUpgradeManager.cs

```diff
@@ -370,13 +220,51 @@ void UpgradePrefabsAndPrefabInstances()
                     conversionLinks.Add(conversionLink);
                     PrefabUtility.RecordPrefabInstancePropertyModifications(go); // record name change
                 }
-
-                conversionLinksPerScene.Add(m_SceneManager.GetScenePath(s), conversionLinks);
+                conversionLinksPerScene.Add(s, conversionLinks);
                 // save changes (i.e. converted prefab instances)
                 EditorSceneManager.SaveScene(scene); 
             }
 
-            // Upgrade Prefab Assets
+            return conversionLinksPerScene;
+        }
+        
+        static List<GameObject> GetAllGameObjects()
+        {
+            var all = (GameObject[])Resources.FindObjectsOfTypeAll(typeof(GameObject));
+            return all.Where(go => !EditorUtility.IsPersistent(go.transform.root.gameObject)
+                && (go.hideFlags & (HideFlags.NotEditable | HideFlags.HideAndDontSave)) == 0).ToList();
+        }
+
+        static void RestoreConvertedCopyReferences(GameObject original, GameObject converted, List<ConversionLink> conversionLinks)
+        {
+            // TODO: if we'll have more components that can reference upgradables, then we'll need to
```

Comment:
Big red hack-flags here.  And what about dolly cart?

Reply:
Definitely. I won't remove this for now, because the reflection based referencing will obsolete this anyway. Until that I leave the hack here.